### PR TITLE
Migrate to tailwindcss

### LINKED ISSUE #526: Convert to Tailwind
How hard to convert this tailwind?

### LINKED ISSUE #545: Resolve flashing while loading the page
While loading the page, there is a flash before the style gets loaded.

### PR DESCRIPTION
**Pull request by ahangarha**
In this PR:

- Bootstrap is completely removed from the project.
- For transitions, we still use CSS modules and Sass for:
  - Lesser changes in the PR
  - Good to have such demonstrations in this project
- I have used the typography plugin for Tailwind CSS for easier styling.

This PR, resolves the flashing issue as well (with production build)

---
Closes #526 
Closes #545 

This change is [Reviewable](https://reviewable.io/reviews/shakacode/react-webpack-rails-tutorial/553)

### CODE REVIEW

File: spec/stimulus/turbo_spec.rb [hunk 1]
```diff
@@ -52,6 +52,7 @@
       fill_in author_field, with: comment.author
       fill_in text_field, with: comment.text
       click_button("Post")
+      sleep(1)
```
Comment (ahangarha) [inline spec/stimulus/turbo_spec.rb:55]: Without forcing some delay, I hardly could get this test pass locally.

File: client/app/assets/styles/application.css [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  @apply text-3xl font-medium mt-5 mb-2.5;
+}
+
+h3 {
+  @apply text-xl font-medium mt-5 mb-2.5
+}
+
+a {
+  @apply text-sky-700
+}
```
Comment (ahangarha) [inline client/app/assets/styles/application.css:20]: This is a violation of the utility approach for styling in Tailwind CSS, I did this to simplify the work.

Should I apply them directly on the elements?

File: package.json [hunk 3]
```diff
@@ -87,9 +84,10 @@
     "sass": "^1.58.3",
     "sass-loader": "^13.3.2",
     "sass-resources-loader": "^2.2.5",
```
Comment (Judahmeek) [inline package.json:86]: You should also be able to remove `sass-resources-loader`

**Review by Judahmeek — CHANGES_REQUESTED** [review]
Overall, it looks really good.

I've just got one small change to request: please remove `sass-resources-loader` if possible.

File: package.json (hunk 3 above)
Comment (ahangarha) [reply package.json]: Done

Comment (ahangarha) [reply package.json:86]: @Judahmeek 
We need to keep `sass-resources-loader` to be used in this project. We want to demonstrate it. By changing to Tailwind, its old usage is no longer relevant, but I should be able to load another sass file.

So, this PR should remain open till I get it to work.

**Comment by ahangarha** [comment]
@justin808 I think https://github.com/shakacode/react-webpack-rails-tutorial/pull/553/commits/384e1c36b88b173cab69b0418a864ddb7ad57d17 satisfies your request regarding keeping `sass-resources-loader` in the project.

**Pull request merged**